### DISCUSSION TITLE
Cancel Interactive Evaluation in FSI

### DISCUSSION THREAD

**Discussion by mathias-brandewinder** [ionide/ionide-vscode-fsharp#1260] [Ideas]
FSI in Visual Studio has a feature, "Cancel Interactive Evaluation", triggered by Ctrl+Break or right-click on the F# Interactive window menu.

This is useful to interrupt a long-running process, without losing what is in memory in the session. As an example, I might want to interrupt a long calculation because I realized there was an error, without restarting the whole session and needing to reload data.

It would be nice to offer that feature, perhaps through an additional command, perhaps using a similar shortcut.

[a comment by an automated account is not shown]

**Comment by baronfel** [comment]
Do you know of a way to perform this cancellation from the terminal running dotnet FSI itself? We only spawn a terminal so if there's not a way to do it via keystrokes then we might be out of luck.

**Comment by mathias-brandewinder** [comment]
Unfortunately, I do not. The `Ctrl+Break` shortcut in Visual Studio doesn't seem to do anything anywhere else. I tried `Ctrl+C`, but that seems to just stop the entire FSI session.

**Reply by tendersoft-mjb** [reply]
If the code is run in Terminal then executing "#quit;;" (without quotes) should end FSI evaluation. However, if run in Output window then this does not work since, as expected, Output window does not accept input.

It should be possible to somehow specify a template/post-command for running F# code/scripts, so a "#quit;;" statement could be appended at the end.

**Comment by goswinr** [comment]
I think this happened via `Thread.Abort()` , but this is not available anymore in netcore , it might come back though: https://github.com/dotnet/runtime/issues/41291 . 

I am wondering what would be the best way to cancel fsi on netcore programmatically. Is FSAC offering that ?

**Reply by Krzysztof-Cieslak** [reply]
> I am wondering what would be the best way to cancel fsi on netcore programmatically. Is FSAC offering that ?

We currently don't have a way to do anything programmatically. FSI is not hosted in FSAC or any other kind of service that we control, but rather it is just a normally spawned process
